path_login.go

```diff
@@ -138,8 +135,27 @@ func (b *backend) handleLogin(ctx context.Context, req *logical.Request, data *f
 		return logical.ErrorResponse(err.Error()), err
 	}
 
+	// Name for the logical.Alias to set
+	aliasName := roleName
+
 	if cert != "" {
-		if err := validateCert(pk, config, principals); err != nil {
+		principal := data.Get("username").(string)
+
+		if principal == "" {
+			// Select the first valid principal in the certificate, if any listed
```

Comment:
This also breaks the current setup where we are authorized when the roleName matches the certificate principal ? (or am I missing something?)

(The removed line 77)

Reply:
This was fixed. By default we accept the rolename as principal, unless an array of accepted principals is provided.